modules/seqexec/web/shared/src/main/scala/seqexec/web/model/boopickle/package.scala

```diff
@@ -3,8 +3,48 @@
 
 package seqexec.model
 
-import seqexec.model.events.SeqexecEvent
 import _root_.boopickle.Default._
+import monocle.Prism
+import seqexec.model.events.SeqexecEvent
+
+package boopickle {
+
+  @SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements"))
+  trait BooPicklerSyntax {
+    implicit class PicklerPrismOps[A, B <: AnyRef](p: Prism[A, B])(implicit PA: Pickler[A]) {
+      def toPickler: Pickler[B] =
+        new Pickler[B] {
+          override def pickle(obj: B)(implicit state: PickleState): Unit = {
+            state.identityRefFor(obj) match {
+              case Some(idx) =>
+                state.enc.writeInt(-idx)
+                ()
+              case None =>
+                // We need a marker
+                state.enc.writeInt(0)
+                // Use the prism to get a representatative for the A and encode it
+                PA.pickle(p.reverseGet(obj))
+                state.addIdentityRef(obj)
+            }
```

Comment:
Is this related to the change to `Observation.Id`?  Either way, it seems like a good idea to create these instances from the optics.